capstone-frontend/src/components/Home.jsx

```diff
@@ -0,0 +1,304 @@
+import CreateList from './CreateList'
+import '../App.css';
+import '../styles/Home.css'
+
+class Home extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      clubsOwned: [],
+      clubsMember: [],
+      myBookListsMap: new Map(),
+      sharedBookListsMap: new Map(),
+      key: 'Clubs - Owner', // For Bootstrap Tabs
+      screenWidth: window.outerWidth,
+      fetchingClubsOwned: true, // For Spinner
+      fetchingClubsMember: true, // For Spinner
+      fetchingBookListsOwned: true, // For Spinner
+      fetchingBookListsShared: true, // For Spinner
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener("resize", this.resize);
+    this.fetchClubs('own');
+    this.fetchClubs('shared');
+    this.fetchBookLists('own');
+    this.fetchBookLists('shared');
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.resize);
+  }
+
+  resize = () => {
+    this.setState({ screenWidth: window.outerWidth });
+
+    this.fetchBookLists('own');
+    this.fetchBookLists('shared');
+  }
+
+  fetchClubs = async (type) => {
+
+    const userID = window.localStorage.getItem('userID');
+
+    if (type === 'own') {
+      this.setState({ fetchingClubsOwned: true });
+      const clubsOwned = await fetch(`/api/clubs?ownerID=${userID}`).then(resp => resp.json());
+      this.setState({ clubsOwned, fetchingClubsOwned: false });
+    }
+    else if (type = 'shared') {
+      this.setState({ fetchingClubsMember: true });
+      const clubsMember = await fetch(`/api/clubs?memberIDs=${userID}`).then(resp => resp.json());
+      this.setState({ clubsMember, fetchingClubsMember: false });
+    }
+  }
+
+  fetchBookLists = async (type) => {
+
+    const userID = window.localStorage.getItem('userID');
+
+    let bookLists;
+
+    if (type === 'own') {
+      this.setState({ fetchingBookListsOwned: true });
+
+      bookLists = await fetch(`/api/booklist?userID=${userID}`)
+        .then(resp => resp.json())
+        .catch(err => console.log(err));
+
+      if (typeof bookLists === 'undefined') {
+        bookLists = [];
+      }
+    }
+    else if (type === 'shared') {
+      this.setState({ fetchingBookListsShared: true });
+
+      bookLists = await fetch(`/api/booklist?collaboratorsIDs=${userID}`)
+        .then(resp => resp.json())
+        .catch(err => console.log(err));
+
+      if (typeof bookLists === 'undefined') {
+        bookLists = [];
+      }
+    }
+
+    let bookListsMap = new Map();
+
+    await Promise.all(bookLists.map(async bookList => {
+
+      if (bookList.gbookIDs.length === 0) {
+        bookListsMap.set(bookList, [])
+        return;
+      }
+
+      let gBooks = [];
+      let bookIDs;
+
+      // Reduce to one element if on mobile
+      if (this.state.screenWidth <= 425) {
+        bookIDs = bookList.gbookIDs.slice(0, 1);
+      }
+      // Reduce to three elements if on table
+      else if (this.state.screenWidth >= 426 && this.state.screenWidth <= 768) {
```

Comment:
nit: Since this is an **else** if, couldn't you just use the second case (`this.state.screenWidth <= 768`) because we assume that the first statement is true already?

Reply:
Oh, I see. Should I change it? I feel like the current implementation is more readable. Let me know.